crates/ruff/src/rules/pyupgrade/rules/f_strings.rs

```diff
@@ -338,7 +338,10 @@ pub(crate) fn f_strings(
 
     // Avoid refactors that exceed the line length limit.
     let col_offset = template.start() - checker.locator.line_start(template.start());
-    if col_offset.to_usize() + contents.len() > line_length.get() {
+    if contents
+        .split('\n')
+        .any(|line| col_offset.to_usize() + line.len() > line_length.get())
```

Comment:
The `col_offset` is only correct for the first line, all other lines have an offset of zero.
```python
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa = """
{}
{}
{}
""".format(
1,
2,
333333333333333333333,
)
```
In this example we don't suggest the autofix even though we're inside the line limit. This is a false negative for autofixing though, so i'm also fine with leaving a TODO comment and merging

Reply:
Can we only use `col_offset` for the first line, and ignore it for the rest?

```diff
diff --git a/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs b/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs
index d79dbd560..695b3c35a 100644
--- a/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs
+++ b/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs
@@ -338,10 +338,10 @@ pub(crate) fn f_strings(
 
     // Avoid refactors that exceed the line length limit.
     let col_offset = template.start() - checker.locator.line_start(template.start());
-    if contents
-        .lines()
-        .any(|line| col_offset.to_usize() + line.len() > line_length.get())
-    {
+    if contents.lines().enumerate().any(|(idx, line)| {
+        let offset = if idx == 0 { col_offset.to_usize() } else { 0 };
+        offset + line.len() > line_length.get()
+    }) {
         return;
     }
```

Reply:
yep! this should get a comment tough explaining why we do this